### PR TITLE
Initial usage docs

### PR DESCRIPTION
**Pull request by scotttrinh**
📜 Hooray for docs!

[Rendered for easier reading](https://github.com/ca-la/remote-collection/blob/initial-usage-docs/README.md)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -39,7 +33,329 @@ to account for, etc.
 
 `remote-collection` is here to help with that!
 
-### Usage
+## RemoteCollection
+
+Creates a new `RemoteCollection` of a generic `Resource` type. If you pass in
+another `RemoteRemoteCollection<Resource>` to the constructor, it will merge the
```
Comment (skswanke) [inline README.md]: `RemoteRemoteCollection`

Comment (scotttrinh) [reply README.md]: :zap: c8fdf12

File: README.md [hunk 2]
```diff
@@ -39,7 +33,329 @@ to account for, etc.
...
+```
+
+## withListFailure
+
+If requesting the list fails, calling this method will store a `RemoteFailure`
+with the passed `string`.
+
+### Signature
+```ts
+withResourceAt(at: string, id: string, resource: Resource): RemoteCollection<Resource>
```
Comment (skswanke) [inline README.md]: wrong signature in a few places

**Review by skswanke — CHANGES_REQUESTED** [review]
I'd really like to see a section with some common usages / FAQ

**Comment by scotttrinh** [comment]
@skswanke 

> I'd really like to see a section with some common usages / FAQ

There are a bunch of examples to demonstrate the methods. Maybe we can make a small app with redux and show off some common patterns? Seems a bit out of scope for "add some usage docs", though.

If you have a specific use-case in mind, maybe I can figure out some small example to demonstrate?

**Comment by skswanke** [comment]
@scotttrinh Yeah Maybe a demo/toy app is best. I'm just not super confident if I have a problem I'll be able to solve it using this documentation.

**Comment by scotttrinh** [comment]
@skswanke 

> I'm just not super confident if I have a problem I'll be able to solve it using this documentation.

☝️ 

The pain is real. Another pain point of this is that it returns other types that need more documentation. I think it's rare when we don't understand how `RemoteCollection` works, but how does `RemoteData` work? How does `Option` work? I think a comprehensive example would help that, 100%. I'll prioritize getting something more robust and useful out in the next week.

**Review by skswanke — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -10,25 +10,19 @@ pattern is nice, but there is still a lot of room to improve on the ergonomics
 of dealing with remote data. `remote-collection` aims to provide a user-friendly
 experience to the user while keeping 100% type-safety.
```
Comment (alekhinen) [inline README.md]: redundant use of the term `user` in `provide a user-friendly experience to the user`. cut it to `...aims to provide a user-friendly experience while keeping 100% type-safety.`

File: README.md [hunk 4]
```diff
@@ -1,7 +1,7 @@
 # Remote-Collection
-## A TypeScript `RemoteData` Collection class
+A TypeScript `RemoteData` Collection class
 
-### Rationale
+<hr />
 
 There have been a number of articles written about using Algebraic Data Types to
 describe the states that data fetching entails. Replacing all of the
```
Comment (alekhinen) [inline README.md]: `ergonomics patterns` or `ergonomic patterns`? the first would be correct if you're talking about patterns of ergonomics.

File: README.md [hunk 5]
```diff
@@ -39,7 +33,329 @@ to account for, etc.
...
+## RemoteCollection
+
+Creates a new `RemoteCollection` of a generic `Resource` type. If you pass in
+another `RemoteCollection<Resource>` to the constructor, it will merge the
+resources.
+
+### Signature
+```ts
+export default class Collection<Resource extends { [key: string]: any }> {
+  constructor(fromCollection?: RemoteCollection<Resource>) {
```
Comment (alekhinen) [inline README.md]: can you close out the brackets?
e.g. 
```
export default class Collection<Resource extends { [key: string]: any }> {
  constructor(fromCollection?: RemoteCollection<Resource>) {
    ...
  }
  ...
}
```

File: README.md [hunk 6]
```diff
@@ -39,7 +33,329 @@ to account for, etc.
...
+
+### Signature
+```ts
+withListFailure(error: string): Collection<Resource>
+```
+
+## withListFailureAt
+
+Like `withListFailure` but adds the failure only to the stored list of resources
+at the specified key, but doesn't interfer with other keys or the resource map
```
Comment (alekhinen) [inline README.md]: `interfere`.

**Review by alekhinen — COMMENTED** [review]
a few grammatical nits otherwise lgtm

File: README.md (hunk 5 above)
Comment (scotttrinh) [reply README.md]: ⚡️ https://github.com/ca-la/remote-collection/pull/11/commits/8028c2c27a0cc1fbfcd4738b56df16f95ade2081
Changed this up a bit and added some example docs.

File: README.md (hunk 4 above)
Comment (scotttrinh) [reply README.md]: ⚡️ https://github.com/ca-la/remote-collection/pull/11/commits/a3abd52a4a55db79e66f84210e283d2ec17cea9a

Cleaned the opening paragraph up a bit to address the two (excellent) clarification requests here.

Comment (scotttrinh) [reply README.md]: And also https://github.com/ca-la/remote-collection/pull/11/commits/f833bab3f83995cff4d6c529d1cad828d4c95bb3

**Review by alekhinen — APPROVED** [review]

**Pull request merged**